### PR TITLE
[dmlc] Add new port

### PR DESCRIPTION
**Pull request by NancyLi1013**
Dmlc only supports static build. When building dynamic, it will shows:

 ```
Expected Debug,Dynamic crt linkage, but the following libs had invalid crt linkage:

    F:/update/vcpkg/packages/dmlc-core_x86-windows/debug/lib/dmlc.lib: Debug,Static

To inspect the lib files, use:
    dumpbin.exe /directives mylibfile.lib
Expected Release,Dynamic crt linkage, but the following libs had invalid crt linkage:

    F:/update/vcpkg/packages/dmlc-core_x86-windows/lib/dmlc.lib: Release,Static

To inspect the lib files, use:
    dumpbin.exe /directives mylibfile.lib
Found 2 error(s). Please correct the portfile:
    F:\update\vcpkg\ports\dmlc-core\portfile.cmake
-- Performing post-build validation done
Error: Building package dmlc-core:x86-windows failed with: POST_BUILD_CHECKS_FAILED
```
So I added `vcpkg_check_linkage(ONLY_STATIC_LIBRARY ONLY_STATIC_CRT)` in portfile.cmake.

### CODE REVIEW

File: ports/dmlc-core/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,27 @@
+include(vcpkg_common_functions)
+
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY ONLY_STATIC_CRT)
```
Comment (dan-shaw) [inline ports/dmlc-core/portfile.cmake]: We might want to make this a rolling-release port. In https://github.com/dmlc/dmlc-core/pull/547 it appears this problem is fixed with the option of `DMLC_FORCE_SHARED_CRT`. If `VCPKG_CRT_LINKAGE` is `dynamic`, we should set this option.

Also, this port appears to be passing on Linux if we comment out this line.

Comment (NancyLi1013) [reply ports/dmlc-core/portfile.cmake]: @dan-shaw , thanks for your review. I have updated this port. It seems that this port currently doesn't support UWP. I added this message in portfile.cmake. Please help check it again. Thanks.

File: ports/dmlc-core/CONTROL [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Source: dmlc-core
+Version: 2019-08-12
+Homepage: https://github.com/dmlc/dmlc-core
+Description: DMLC-Core is the backbone library to support all DMLC projects, offers the bricks to build efficient and scalable distributed machine learning libraries.
```
Comment (dan-shaw) [inline ports/dmlc-core/CONTROL]: `dmlc-core` has a dependency on `OpenMP` (the osx CI test is unable to find `OpenMP`). We should make a new feature `openmp` and list it in the `Default-Features`.

**Comment by NancyLi1013** [comment]
I noticed that it failed on x86-windows, due to

>  C:/vsts/_work/1/s/buildtrees/dmlc-core/src/5547c00bf1-c597496dc4/include/dmlc/build_config.h
> CMake Error at CMakeLists.txt:106 (configure_file):
>   configure_file Problem configuring file
> 
> 
> -- Could NOT find PythonInterp (missing: PYTHON_EXECUTABLE) 
> -- Configuring incomplete, errors occurred!

But it can be built successfully on my local. So I'm confused about this.

**Pull request merged**